modules/serverless/pages/serving/serving-transport-encryption.adoc

```diff
@@ -0,0 +1,446 @@
+  namespace: knative-serving
+spec:
+  # Other spec fields omitted ...
+  config:
+    certmanager:
+      clusterLocalIssuerRef: |
+        kind: ClusterIssuer
+        name: knative-serving-ca-issuer <1>
+      systemInternalIssuerRef: |
+        kind: ClusterIssuer
+        name: knative-serving-ca-issuer <1>
+    network:
+      cluster-local-domain-tls: Enabled <2>
+      system-internal-tls: Enabled      <3>
+----
++
+<1> Define the `ClusterIssuer` for each feature. The same or individual `ClusterIssuers` can be used.
+<2> Enabling the `cluster-local-domain-tls` feature. They can be enabled/disabled individually.
+<3> Enabling the `system-internal-tls` feature. They can be enabled/disabled individually.
+
+. Apply the `KnativeServing` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. Restart the Controller component if you enabled `cluster-local-domain-tls` or `system-internal-tls`:
++
+[IMPORTANT]
+====
+When either the `cluster-local-domain-tls` or the `system-internal-tls` feature is enabled, the Controller component needs to be restarted to enable the Knative Serving cert-manager integration.
+====
++
+[source,terminal]
+----
+$ oc rollout restart deploy/controller -n knative-serving
+----
+
+. Restart the Activator component if you enabled `system-internal-tls`
++
+[IMPORTANT]
+====
+When the `system-internal-tls` feature is activated, the Activator component needs to be restarted to reconfigure its internal web server, as this is not possible during runtime.
+====
++
+[source,terminal]
+----
+$ oc rollout restart deploy/activator -n knative-serving
+----
+
+
+== Configure trust
+
+When you enable any of the transport encryption features, you must make sure that all clients calling do trust the Certificate Authority (CA) that issues the certificates used for the transport encryption.
+
+There are multiple places where trust needs to be ensured:
+
+* Cluster external client (Browser and/or other application): this is considered out of the scope of {serverlessproductname}.
+* {serverlessproductname} system components (e.g. Activator, Queue-Proxy, Ingress-Controller): see below.
+* Cluster internal client (e.g. a Knative Service or other workload): see below.
+
+=== Configuring trust for {serverlessproductname} Serving components and Knative Services  [[configuring_trust]]
+
+For {serverlessproductname} Serving components and Knative Services to trust the CA that issues certificates, you can create a `ConfigMap` in the following namespaces with the label `networking.knative.dev/trust-bundle: true`:
+
+* `knative-serving`: for the system components of {serverlessproductname} Serving.
+* `knative-serving-ingress`: for the ingress layer of {serverlessproductname} Serving.
+* `istio-system` or your own {smproductshortname} namespace: when the {smproductshortname} integration is enabled.
+* `<workload namespaces>`: each namespace where a Knative Service is installed (or could be installed, to avoid initial downtime).
```

Comment:
I have only seen using the `networking.knative.dev/trust-bundle: true` label for ConfigMaps in knative-serving (for Activator) and knative-serving-ingress (for net-kourier-controller) namespaces. My understanding is that this is used to build trust, i.e. whom to trust when sending a request forward. Does it make sense to have it in the workload namespace then? Can you point me to the code that uses it? (would be useful for writing any further tests too.)

Reply:
You are right, it's not used. At least not at the moment. It would be necessary e.g. when the QP would validate the client-certificate of activator - but as this is not there yet (and unclear if it ever will), I removed it here.